src/lib/state/DY.Lib.State.Typed.fst

```diff
@@ -19,8 +19,85 @@ let mk_local_state_instance #a #format tag = {
   format;
 }
 
+val typed_state_pred_label_pred:
+  #a:Type0 -> {|parseable_serializeable bytes a|} ->
+  (principal -> string -> state_id -> a -> prop) ->
+  (principal -> string -> state_id -> bytes -> prop)
+let typed_state_pred_label_pred #a #ps_a p prin sess_id tag content =
+  match parse a content with
+  | None -> False
+  | Some parsed_content ->
+    p prin sess_id tag parsed_content
+
+val typed_state_pred_label:
+  #a:Type0 -> {|parseable_serializeable bytes a|} ->
+  (principal -> string -> state_id -> a -> prop) ->
+  label
+let typed_state_pred_label p =
+  tagged_state_pred_label (typed_state_pred_label_pred p)
+
+val was_corrupt:
+  #a:Type0 -> {|parseable_serializeable bytes a|} ->
+  trace ->
+  principal -> string -> state_id -> a ->
+  prop
+let was_corrupt #a #ps tr prin tag sid content =
+  DY.Core.was_corrupt tr prin sid (serialize tagged_state {
+    tag;
+    content = serialize a content;
+  })
```

Comment:
Why don't we have a tagged version of `was_corrupt` which we could use here?
I think, this would be more consistent with the rest of the functions here that are all based on the tagged versions. (And also every function would exist on each level -- core, tagged, typed)

Reply:
Done.